core-tests/src/test/java/org/mule/runtime/core/privileged/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -188,13 +190,21 @@ public void before() throws MuleException {
   public void after() throws MuleException {
     flow.stop();
     flow.dispose();
+
+    if (messageProcessor != null) {
+      stopIfNeeded(messageProcessor);
+      disposeIfNeeded(messageProcessor, LOGGER);
+
+      messageProcessor = null;
+    }
   }
 
   @Test
   public void testMPChain() throws Exception {
     DefaultMessageProcessorChainBuilder builder = new DefaultMessageProcessorChainBuilder();
     builder.chain(getAppendingMP("1"), getAppendingMP("2"), getAppendingMP("3"));
-    assertEquals("0123", process(builder.build(), getTestEventUsingFlow("0")).getMessage().getPayload().getValue());
+    messageProcessor = builder.build();
+    assertEquals("0123", process(messageProcessor, getTestEventUsingFlow("0")).getMessage().getPayload().getValue());
```

Comment:
do you feel like migrating these assertions to hamcrest while you are at it?

Reply:
Done.